### ISSUE TITLE
GbrasNet Layers Pytorch Conversion

### ISSUE THREAD

**Issue by Gil32610** [Gil32610/cryptography-miniproject#1]
Oi @HeloisaJ !

Procurei por aqui as layers equivalentes para o modelo GBRASNET.

São as seguintes:

| TensorFlow                                          | PyTorch                                                                                   |
|----------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------|
| `Input(shape=(256,256,1))`                                           | No explicit input layer; first layer (`Conv2d`) defines input channels. TF uses `(B,H,W,C)`, PyTorch `(B,C,H,W)` |
| `Conv2D(30, (5,5), strides=(1,1), padding='same', activation=Tanh3, use_bias=True, weights=[srm_weights,biasSRM], trainable=False)` | `nn.Conv2d(in_channels=1, out_channels=30, kernel_size=5, stride=1, padding=2, bias=True)` <br> Load custom weights: `conv.weight.data = torch.tensor(srm_weights.transpose(3,2,0,1))` <br> Freeze: `param.requires_grad = False` <br> Activation: `class Tanh3(nn.Module): forward(x): return 3*torch.tanh(x)` |
| `BatchNormalization(momentum=0.2, epsilon=0.001, center=True, scale=False)` | `nn.BatchNorm2d(num_features=30, eps=0.001, momentum=0.2, affine=False)` <br> (`affine=False` disables learnable scale & bias) |
| `DepthwiseConv2D(1)`                                                 | `nn.Conv2d(in_channels=30, out_channels=30, kernel_size=3, stride=1, padding=1, groups=30, bias=True)` <br> (`groups=in_channels` for depthwise) |
| `SeparableConv2D(30, (3,3), padding='same', activation='elu', depth_multiplier=3)` | Custom class: <br> `class SeparableConv2d(nn.Module):` <br> Depthwise: `nn.Conv2d(in_channels, in_channels*depth_multiplier, kernel_size=3, padding=1, groups=in_channels, bias=False)` <br> Pointwise: `nn.Conv2d(in_channels*depth_multiplier, out_channels, kernel_size=1, bias=True)` <br> Activation: `nn.ELU()` |
| `AveragePooling2D((2,2), strides=(2,2))`                             | `nn.AvgPool2d(kernel_size=2, stride=2, padding=0)` <br> (`padding=0` for default, can adjust if needed) |
